File: N개의_최소공배수.py
def gcd(a,b):
    while(True):
        large = max(a,b)
        small = min(a,b)
        if large%small == 0:
            return small
        a=large-small
        b=small

def solution(arr):
    a = arr.pop()
    while(arr):
        b = arr.pop()
        d = gcd(a,b)
        a = a*b//d
    return a

File: golang/Level 2/N개의_최소공배수.go
func gcd(a int, b int) int {
    var large, small int
    
    for {
        if a > b {
            large = a
            small = b
        } else {
            large = b
            small = a
        }
        if large % small == 0 {
            return small
        }
        
        a = large - small
        b = small
    }
}

func solution(arr []int) int {
    a := arr[len(arr)-1]
    for i := len(arr)-2; i > -1; i-- {
        b := arr[i]
        d := gcd(a, b)
        a = int(a * b / d)
    }
    return a
}
